File: mobile_app/lib/config/countries.dart
class CountryConfig {
  final String code; // ISO 3166-1 alpha-2 (e.g., 'DE', 'TR')
  final String name;
  final String currency; // ISO 4217 (e.g., 'EUR', 'TRY')
  final String symbol; // e.g., '€', '₺'
  final String dialCode;

  const CountryConfig({
    required this.code,
    required this.name,
    required this.currency,
    required this.symbol,
    required this.dialCode,
  });
}

class SupportedCountries {
  static const Map<String, CountryConfig> countries = {
    'DE': CountryConfig(
      code: 'DE',
      name: 'Deutschland',
      currency: 'EUR',
      symbol: '€',
      dialCode: '+49',
    ),
    'TR': CountryConfig(
      code: 'TR',
      name: 'Türkiye',
      currency: 'TRY',
      symbol: '₺',
      dialCode: '+90',
    ),
    // Add more countries here
  };

  static const String defaultCountryCode = 'DE';
  static const String defaultCurrency = 'EUR';

  static CountryConfig getCountryConfig([String? countryCode]) {
    return countries[countryCode] ?? countries[defaultCountryCode]!;
  }

  static String getCurrencyForCountry([String? countryCode]) {
    if (countryCode == null) return defaultCurrency;
    return countries[countryCode]?.currency ?? defaultCurrency;
  }

  static String getSymbolForCurrency(String currencyCode) {
    for (final country in countries.values) {
      if (country.currency == currencyCode) {
        return country.symbol;
      }
    }
    return currencyCode; // Fallback to characters if no symbol is found
  }
}

File: admin_portal/src/lib/config/countries.ts
export interface CountryConfig {
    code: string; // ISO 3166-1 alpha-2 (e.g., 'DE', 'TR')
    name: string;
    currency: string; // ISO 4217 (e.g., 'EUR', 'TRY')
    symbol: string; // e.g., '€', '₺'
    dialCode: string; // e.g., '+49', '+90'
    locale: string;
}

export const SUPPORTED_COUNTRIES: Record<string, CountryConfig> = {
    DE: {
        code: 'DE',
        name: 'Deutschland',
        currency: 'EUR',
        symbol: '€',
        dialCode: '+49',
        locale: 'de-DE',
    },
    TR: {
        code: 'TR',
        name: 'Türkiye',
        currency: 'TRY',
        symbol: '₺',
        dialCode: '+90',
        locale: 'tr-TR',
    },
    // Add more countries here as needed
};

// Default fallback for legacy or unspecified cases
export const DEFAULT_COUNTRY_CODE = 'DE';
export const DEFAULT_CURRENCY = 'EUR';

export function getCountryConfig(countryCode: string = DEFAULT_COUNTRY_CODE): CountryConfig {
    return SUPPORTED_COUNTRIES[countryCode] || SUPPORTED_COUNTRIES[DEFAULT_COUNTRY_CODE];
}

export function getCurrencyForCountry(countryCode?: string): string {
    if (!countryCode) return DEFAULT_CURRENCY;
    return SUPPORTED_COUNTRIES[countryCode]?.currency || DEFAULT_CURRENCY;
}

export function getSymbolForCurrency(currencyCode: string): string {
    const country = Object.values(SUPPORTED_COUNTRIES).find((c) => c.currency === currencyCode);
    return country?.symbol || currencyCode; // Fallback to the text code if symbol not found
}
